config.go

```diff
@@ -12,23 +14,54 @@ type Config struct {
 	Packages map[string]Package `yaml:"packages"`
 }
 
+// orderedConfig is used internally to check for alphabetical ordering in the YAML
+// configuration. A yaml.MapSlice perservers ordering of keys: https://godoc.org/gopkg.in/yaml.v2#MapSlice
+type orderedConfig struct {
+	URL      string        `yaml:"url"`
+	Packages yaml.MapSlice `yaml:"packages"`
+}
+
 // Package details the options available for each repo
 type Package struct {
 	Repo string `yaml:"repo"`
 }
 
+// ensureAlphabetical checks that the packages are listed alphabetically in the configuration.
+func ensureAlphabetical(data []byte) bool {
+	var c orderedConfig
+
+	if err := yaml.Unmarshal(data, &c); err != nil {
+		return false
+	}
+
+	packageNames := make([]string, len(c.Packages))
+	for _, v := range c.Packages {
+		name, ok := v.Key.(string)
+		if !ok {
+			return false
+		}
+		packageNames = append(packageNames, name)
+	}
+
+	return sort.StringsAreSorted(packageNames)
+}
+
 // Parse takes a path to a yaml file and produces a parsed Config
-func Parse(path string) (Config, error) {
+func Parse(path string) (*Config, error) {
 	var c Config
 
 	data, err := ioutil.ReadFile(path)
 	if err != nil {
-		return c, err
+		return nil, err
 	}
 
 	if err := yaml.Unmarshal(data, &c); err != nil {
-		return c, err
+		return nil, err
+	}
+
+	if !ensureAlphabetical(data) {
```

Comment:
While that's a good idea, map iteration is random so the checks would be inconsistent. This is why I resorted to using the `yaml.MapSlice` since that is ordered.

Is that fair?